gen/qr.hoon

```diff
@@ -0,0 +1,359 @@
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+::
+::  QR encode a tape, and print it as a %tang.
+::
+::  usage:  1) place this file in gen/qr.hoon
+::          2) at the dojo, run `+qr "tape to encode"`
+::
+::  currently, QR version 5L, with and mask 0, are hardcoded.
+::             this limits messages to < 106 characters long,
+::             and the code produced may be difficult to scan
+::
+::  TODO:  implement the other 7 masks
+::         penalty scoring
+::         move the "two '█' per module" logic up to pretty-print
+::
+::  TOC::...
+::       :1: main generator
+::       :2: types
+::       :3: main logic
+::       :4: qr formatting helpers
+::       :5: message encoding helpers
+::       :6: long constants
+::
+::  implemented with much help from the following sites:
+::    https://urbit.org/docs/
+::    https://www.thonky.com/qr-code-tutorial/
+::    https://en.wikiversity.org/wiki/Reed%E2%80%93Solomon_codes_for_coders
+::
+::::  /~somrem-bosdes/home/gen/qr/hoon  ::::::::::::::::::::::::::::::::::::::::
+::
+::::::::::::::::::::::::::::::::::::::::::::::::1: main generator
+::
+:-  %say
+|=  [* [msg=tape ~] ~ ~]
+::
+?:  (gth (lent msg) 106)
+  :-  %tape
+      "Tape must be 106 characters or less, yours is {<(lent msg)>} characters"
+=-
+:-  %tang
+  %-  pretty-print.qr
+  %-  encode.qr  msg
+::
+^=  qr  |%
+::
+::::::::::::::::::::::::::::::::::::::::::::::::2: types
+::
+::  TODO type this better, will ease implementation of new features
+::
+++  grid  (list (list ?))
```

Comment:
Thanks, that does sound better. It would make it a lot more efficient to use things like ▄▀ in `pretty-print`, so you could look up "neighborhoods" of modules quickly, instead of linearly going along in my list of list of loobeans...